Classes/Service/ExtconfService.php

```diff
@@ -22,49 +23,151 @@ public static function getInstance()
         return GeneralUtility::makeInstance(ExtconfService::class);
     }
 
+    /**
+     * Creates a new ES indice for every indece Indexer combination in the config. This keeps the configuration simple even if it is not the real index structure in elasticsearch.
+     * This is necessary because since elastic 7.0 one indice can only have one document type.
+     *
+     * @return array
+     */
+    protected static function getElasticsearchIndices()
+    {
+        $indices = [];
+        foreach ($GLOBALS['TYPO3_CONF_VARS']['EXTCONF']['searchable']['indices'] as $indexKey => $index) {
+            if (!isset($index['typo3_language'])) {
+                throw new \Exception('Please set the "typo3_language" for the index ' . $indexKey, 1719935622);
+            }
+
+            if (isset($index['indexer']) && !empty($index['indexer'])) {
+                $indexer = $index['indexer'];
+            } else {
+                // If no indexer are set use all defined indexers
+                $indexer = array_keys($GLOBALS['TYPO3_CONF_VARS']['EXTCONF']['searchable']['indexers']);
+            }
+            foreach ($indexer as $key) {
+                $indices[$indexKey . '_' . $key] = $index;
+                $indices[$indexKey . '_' . $key]['indexer'] = $key;
+                $indices[$indexKey . '_' . $key]['configIndex'] = $indexKey;
+            }
+        }
+
+        return $indices;
+    }
+
     /**
      * Returns all available indices
      *
      * @return array
      */
     public static function getIndices()
     {
-        $indicesConfiguration = $GLOBALS['TYPO3_CONF_VARS']['EXTCONF']['searchable']['indices'];
+        $indicesConfiguration = ExtconfService::getElasticsearchIndices();
+
+        return array_keys($indicesConfiguration);
+    }
+
+
+    /**
+     * Returns the index language for a given name, if set. Otherwise throws an error so no invalid indices are created
+     *
+     * @param  int $language
+     * @return array $indices
+     */
+    public static function getIndecesByLanguage($language = 0)
```

Comment:
```suggestion
    public static function getIndecesByLanguage(int $language = 0)
```

We should prefer native types.